internal/eval/eval.go

```diff
@@ -2782,3 +2783,153 @@ func evalSELECT(args []string, store *dstore.Store) []byte {
+	if errWithMessage != nil {
+		return errWithMessage
+	}
+
+	path := args[1]
+
+	jsonData := obj.Value
+	// Parse the JSONPath expression
+	expr, err := jp.ParseString(path)
+
+	if err != nil {
+		return diceerrors.NewErrWithMessage("invalid JSONPath")
+	}
+
+	isIncrFloat := false
+
+	for i, r := range args[2] {
+		if !unicode.IsDigit(r) && r != '.' && r != '-' {
+			if i == 0 {
+				return diceerrors.NewErrWithFormattedMessage("-ERR expected value at line 1 column %d", i+1)
+			}
+			return diceerrors.NewErrWithFormattedMessage("-ERR trailing characters at line 1 column %d", i+1)
+		}
+		if r == '.' {
+			isIncrFloat = true
+		}
+	}
+	var incrFloat float64
+	var incrInt int64
+	if isIncrFloat {
+		incrFloat, err = strconv.ParseFloat(args[2], 64)
+		if err != nil {
+			return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
+		}
+	} else {
+		incrInt, err = strconv.ParseInt(args[2], 10, 64)
+		if err != nil {
+			return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
+		}
+	}
+	results := expr.Get(jsonData)
+
+	if len(results) == 0 {
+		respString := "[]"
+		return clientio.Encode(respString, false)
+	}
+
+	resultArray := make([]string, 0, len(results))
+
+	if path == defaultRootPath {
+		switch utils.GetJSONFieldType(jsonData) {
+		case utils.NumberType:
+			oldVal := jsonData.(float64)
+			if isIncrFloat {
+				jsonData = oldVal + incrFloat
+			} else {
+				jsonData = oldVal + float64(incrInt)
+			}
+			resultArray = append(resultArray, formatFloat(jsonData.(float64), isIncrFloat))
+		case utils.IntegerType:
+			if isIncrFloat {
+				oldVal := jsonData.(float64)
+				jsonData = oldVal + incrFloat
+				resultArray = append(resultArray, formatFloat(jsonData.(float64), isIncrFloat))
+			} else {
+				oldVal := jsonData.(int64)
+				jsonData = oldVal + incrInt
+				resultArray = append(resultArray, fmt.Sprintf("%d", jsonData))
+			}
+		default:
+			resultArray = append(resultArray, "null")
+		}
+	} else {
+		// Execute the JSONPath query
+
+		expr.MustModify(jsonData, func(value any) (interface{}, bool) {
+			switch utils.GetJSONFieldType(value) {
+			case utils.NumberType:
+				oldVal := value.(float64)
+				var newVal float64
+				if isIncrFloat {
+					newVal = oldVal + incrFloat
+				} else {
+					newVal = oldVal + float64(incrInt)
+				}
+				resultArray = append(resultArray, formatFloat(newVal, isIncrFloat))
+				return newVal, true
+			case utils.IntegerType:
+				if isIncrFloat {
+					oldVal := value.(float64)
+					newVal := oldVal + incrFloat
+					resultArray = append(resultArray, formatFloat(newVal, isIncrFloat))
+					return newVal, true
+				} else {
+					oldVal := value.(int64)
+					newVal := oldVal + incrInt
+					resultArray = append(resultArray, fmt.Sprintf("%d", newVal))
+					return newVal, true
+				}
+			default:
+				resultArray = append(resultArray, "null")
+				return value, false
+			}
+		})
+	}
```

Comment:
There is a lot of code duplication here.

You can replace this with:
```go
	if path == defaultRootPath {
		newValue, resultString, modified := incrementValue(jsonData, isIncrFloat, incrFloat, incrInt)
		if modified {
			jsonData = newValue
		}
		resultArray = append(resultArray, resultString)
	} else {
		_, err := expr.Modify(jsonData, func(value any) (interface{}, bool) {
			newValue, resultString, modified := incrementValue(value, isIncrFloat, incrFloat, incrInt)
			resultArray = append(resultArray, resultString)
			return newValue, modified
		})
		if err != nil {
			return diceerrors.NewErrWithMessage("invalid JSONPath")
		}
	}
```

where `incrementValue` is implemented as follows:
```go
func incrementValue(value any, isIncrFloat bool, incrFloat float64, incrInt int64) (interface{}, string, bool) {
	switch utils.GetJSONFieldType(value) {
	case utils.NumberType:
		oldVal := value.(float64)
		var newVal float64
		if isIncrFloat {
			newVal = oldVal + incrFloat
		} else {
			newVal = oldVal + float64(incrInt)
		}
		resultString := formatFloat(newVal, isIncrFloat)
		return newVal, resultString, true
	case utils.IntegerType:
		if isIncrFloat {
			oldVal := float64(value.(int64))
			newVal := oldVal + incrFloat
			resultString := formatFloat(newVal, isIncrFloat)
			return newVal, resultString, true
		} else {
			oldVal := value.(int64)
			newVal := oldVal + incrInt
			resultString := fmt.Sprintf("%d", newVal)
			return newVal, resultString, true
		}
	default:
		return value, "null", false
	}
}
```